puppet-enterprise-cluster/azuredeploy.json

```diff
@@ -0,0 +1,506 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "adminUsername": {
+      "type": "string",
+      "metadata": {
+        "description": "User name for the Virtual Machine. Used for both SSH and RDP access."
+      }
+    },
+    "adminPassword": {
+      "type": "securestring",
+      "metadata": {
+        "description": "Password for the Virtual Machine. Used for both SSH and RDP access."
+      }
+    },
+    "windowsAgentCount": {
+      "type": "int",
+      "metadata": {
+        "description": "Number of Windows Puppet Agents to deploy. Assumes Windows Server 2012 R2."
+      },
+      "defaultValue": 1
+    },
+    "linuxAgentCount": {
+      "type": "int",
+      "metadata": {
+        "description": "Number of Linux Puppet Agents to deploy. Assumes Ubuntu 14.04-LTS."
+      },
+      "defaultValue": 1
+    },
+    "vmSize": {
+      "type": "string",
+      "metadata": {
+        "description": "Size of Puppet agent VMs"
+      },
+      "defaultValue": "Standard_D1_v2"
+    }
+  },
+  "variables": {
+    "puppetAgentCount": "[add(parameters('windowsAgentCount'),parameters('linuxAgentCount'))]",
+    "newStorageAccountName": "[concat('str',uniqueString(resourceGroup().id, deployment().name))]",
+    "dnsNameForPublicIP": "[concat('dns',uniqueString(resourceGroup().id, deployment().name))]",
+    "imagePublisher": "Puppet",
+    "imageOffer": "Puppet-Enterprise",
+    "imageSku": "2016-1",
+    "OSDiskName": "puppetMasterDisk",
+    "nicName": "puppetmasterNic",
+    "addressPrefix": "10.0.0.0/16",
+    "subnetName": "Subnet",
+    "subnetPrefix": "10.0.0.0/24",
+    "storageAccountType": "Standard_LRS",
+    "publicIPAddressName": "myPublicIP",
+    "publicIPAddressType": "Dynamic",
+    "vmStorageAccountContainerName": "vhds",
+    "puppetMasterVmName": "puppetmaster",
+    "vmName": "peagent",
+    "virtualNetworkName": "vnet1",
+    "vnetID": "[resourceId('Microsoft.Network/virtualNetworks',variables('virtualNetworkName'))]",
+    "subnetRef": "[concat(variables('vnetID'),'/subnets/',variables('subnetName'))]",
+    "publicIpId": "[resourceId(resourceGroup().Name,concat('Microsoft.Network','/','publicIPAddresses'),variables('publicIPAddressName'))]"
+  },
+  "resources": [
+    {
+      "type": "Microsoft.Storage/storageAccounts",
+      "name": "[variables('newStorageAccountName')]",
+      "apiVersion": "2015-05-01-preview",
+      "location": "[resourceGroup().location]",
+      "properties": {
+        "accountType": "[variables('storageAccountType')]"
+      }
+    },
+    {
+      "apiVersion": "2015-05-01-preview",
```

Comment:
I think 2016-03-30 is the latest for network as well as compute. @bmoore-msft Can you confirm?